### PR TITLE
Architecture basics

### PR DESCRIPTION
**Pull request by gmelodie**
Add general skeleton for `server.py` and `client.py` according to general new architecture discussed.

**Testing:** in the root folder do `unvicorn howdoi.server:app` to run the server. With the server running use `python3 client.py` to issue a query. It takes exactly a normal howdoi query, sends it to the server, prints the answer and exits.

### CODE REVIEW

File: howdoi/client.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+SERVER_URL = os.getenv("HOWDOI_SERVER_URL",
+                       default="http://localhost:8000/plugins/")
+
+
+if __name__ == '__main__':
+    query = input("Type query: ")
+    response = requests.post(SERVER_URL, json={"query_str": query})
+    print(response.json()["answer_str"])
+    # for key, value in response.json().items():
+    #     print(key, value)
```
Comment (teymour-aldridge) [inline howdoi/client.py:14]: Minor style thing
```suggestion
    #     print(key, value)

```

File: howdoi/server.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+
+@app.post('/plugins/{plugin}', response_model=Response)
+async def plugin(plugin: str, request: Request):
+    answer_str = _get_answer(request.query_str)
+
+    return {
+        "answer_str": answer_str,
+        "plugin": plugin,
+    }
```
Comment (teymour-aldridge) [inline howdoi/server.py:48]: Again, minor style thing
```suggestion
    }

```

File: howdoi/test_server.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    }
+    assert good_response.status_code == 200
+
+    assert bad_response.status_code == 422 # unprocessable
+
+
+def test_root():
+    """Test root / endpoint"""
+    response = client.get('/')
+    assert response.status_code == 404
```
Comment (teymour-aldridge) [inline howdoi/test_server.py:72]: Again, minor style thing
```suggestion
    assert response.status_code == 404

```

File: howdoi/server.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+app = FastAPI(
+    title="Howdoi Clients",
+    description="Howdoi's API for clients",
+    version="1",
+)
+
+
+def _get_answer(query_str):
+    # TODO: maybe this wouldn't be needed in the future,
+    #       it would be done in the client
```
Comment (teymour-aldridge) [inline howdoi/server.py:26]:
```suggestion
    #       it would be done by the client
```

**Comment by IamCathal** [comment]
A bit of personal preference but I like to do my tests like this since it's more concise
```python
 queries = [
        { "content":"", "want": 422 },
        { "content":"", "want": 422 },
        { "content":"test", "want": 200 },
        { "content":"test2", "want": 200 },
    ]

    for testCase in queries:

        response = client.post(
            "/plugins",
            headers={},
            json={
                "query_str": testCase.content,
                "other_content": "some other content"
            },
        )
        assert testCase["want"] == response.status_code 
````

**Comment by gmelodie** [comment]
We ended up getting some of the ideas for the new architecture and are implementing them differently. Closing this.

**Pull request closed without merging**